### ISSUE TITLE
[BUG] Clients registered with `AddAzureClients` are not disposed

### ISSUE THREAD

**Issue by tom-wolfe** [Azure/azure-sdk-for-net#33131]
### Library name and version

Microsoft.Extensions.Azure 1.6.0

### Describe the bug

Not 100% if bug or intended behaviour, but could do with a documentation clarification if not.

When using the Azure Client Builder/Factory, the registered clients aren't disposed along with the `IServiceProvider`.

### Expected behavior

When calling `provider.Dispose()` in the provided code snippet, I would expect the resolved client instance to be disposed, because the [Dependency Injection guidelines](https://learn.microsoft.com/en-us/dotnet/core/extensions/dependency-injection-guidelines#disposal-of-services) specify not to dispose of services resolved from the container manually.

### Actual behavior

The resolved client is left undisposed because the object that actually gets registered with the `IServiceCollection` is the `ClientRegistration` object, which doesn't implement `IDisposable`.

### Reproduction Steps

```cs
using Microsoft.Extensions.Azure;
using Microsoft.Extensions.DependencyInjection;

public class Program
{
    public static void Main()
    {
        var services = new ServiceCollection();
        services.AddAzureClients(builder => builder
            .AddClient<MyDisposableClient, object>(_ => new MyDisposableClient()).WithName("TestClient")
        );
        var provider = services.BuildServiceProvider();
        
        var factory = provider.GetRequiredService<IAzureClientFactory<MyDisposableClient>>();
        
        var client = factory.CreateClient("TestClient");
        client.Write("Client created successfully.");

        provider.Dispose(); // <-- Expected dispose message to be printed at this point.
        Console.WriteLine("Provider disposed");

        Console.ReadLine();
    }

    class MyDisposableClient : IDisposable
    {
        public void Write(string message) => Console.WriteLine(message);
        public void Dispose() => Write("Client disposed");
    }
}
```

### Environment

IDE: Visual Studio Professional 2022 (64-bit) (17.4.2)
```
.NET SDK:
 Version:   7.0.101
 Commit:    bb24aafa11

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.22621
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\7.0.101\

Host:
  Version:      7.0.1
  Architecture: x64
  Commit:       97203d38ba
```

**Comment by jsquire** [comment]
Hi @tom-wolfe.  Thank you for reaching out and we regret that you're experiencing difficulties.   We'll investigate.

**Comment by jsquire** [comment]
//cc: @JoshLove-msft

**Comment by jsquire** [comment]
After investigation, this is a bug and we're working on a fix.

**Comment by jsquire** [comment]
This has been addressed by #34154.  Release is currently planned for early March, tracked by #34186.   Please be aware that release dates may change.

I'm going to close this out.  Please feel free to unresolve if you'd like to continue the discussion.
